UnhollowerBaseLib/ClassInjector.cs

```diff
@@ -773,46 +774,44 @@ private static Type NativeType(this Type type)
         private delegate void ClassInitDelegate(Il2CppClass* klass);
         private static ClassInitDelegate ourClassInitMethod;
 
-        private static ClassInitDelegate FindClassInitMethod()
+        private static readonly MemoryUtils.SignatureDefinition[] classInitSignatures =
         {
-            var lib = LoadLibrary("GameAssembly.dll");
-            var entryPointAddress = GetProcAddress(lib, nameof(IL2CPP.il2cpp_object_new));
-            LogSupport.Trace($"il2cpp_object_new: {entryPointAddress}");
-
-            var entrypointTargets = XrefScannerLowLevel.JumpTargets(entryPointAddress).ToArray();
-
-            IntPtr objectNewAllocSpecificAddress;
-
-            switch (entrypointTargets.Length)
+            new MemoryUtils.SignatureDefinition
             {
-                case 1:
-                {
-                    var objectNewAddress = entrypointTargets.Single();
-                    LogSupport.Trace($"Object::New: {objectNewAddress}");
-
-                    objectNewAllocSpecificAddress = XrefScannerLowLevel.JumpTargets(objectNewAddress).Single();
-
-                    break;
-                }
-
-                case 2:
-                {
-                    objectNewAllocSpecificAddress = entrypointTargets.First();
-                    break;
-                }
+                pattern = "\xE8\x00\x00\x00\x00\x0F\xB7\x47\x28\x83",
+                mask = "x????xxxxx",
+                xref = true
+            },
+            new MemoryUtils.SignatureDefinition
+            {
+                pattern = "\xE8\x00\x00\x00\x00\x0F\xB7\x47\x48\x48",
+                mask = "x????xxxxx",
+                xref = true
+            }
+        };
 
-                default:
-                {
-                    throw new NotSupportedException("Failed to find Class::Init, please create an issue and report your unity version");
-                }
+        private static ClassInitDelegate FindClassInitMethod()
+        {
+            ProcessModule gameAssembly = Process.GetCurrentProcess()
+                .Modules.OfType<ProcessModule>()
+                .Single((x) => x.ModuleName == "GameAssembly.dll");
+            void* pClassInit = (void*)0;
+            foreach(var signature in classInitSignatures)
+            {
+                pClassInit = MemoryUtils.FindSignatureInModule(gameAssembly, signature);
+                if (pClassInit != (void*)0) break;
             }
 
-            LogSupport.Trace($"Object::NewAllocSpecific: {objectNewAllocSpecificAddress}");
+            if (pClassInit == (void*)0)
+            {
+                // WARN: There might be a race condition with il2cpp_class_has_references
+                LogSupport.Warning("Signatures exhausted, settling with il2cpp_class_has_references");
```

Comment:
Maybe a more specific warning message could be useful? Perhaps mention that scanning for `Class::Init` failed. Otherwise, it'd probably seem a bit cryptic what the signatures are and why `il2cpp_class_has_references` is used.

Reply:
Changed the log message to say that signatures for `Class::Init` have been exhausted and `il2cpp_class_has_references` will be used as a substitute.
I'm also throwing an exception now if `il2cpp_class_has_references` cannot be found to direct people to open issues with their unity version & game in case `il2cpp_class_has_references` is missing in a previous/future version if libil2cpp